### PR TITLE
Update to .NET 7 SDK

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Updating to latest SDK to get build improvements (including the check that resulted in the code update in this PR). 

This does not change the target framework for our extension - that will still be net6.

### CODE REVIEW

**Comment by chlafreniere** [comment]
Is there an existing issue that this fixes?

File: builds/azure-pipelines/template-steps-build-test.yml [hunk 1]
```diff
@@ -174,13 +174,19 @@ steps:
     GdnPublishTsaConfigFile: '$(Build.SourcesDirectory)\builds\TSAConfig.gdntsa'
   condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'), eq(variables['TSA_UPLOAD'], 'true'))
 
-# 5.0 isn't supported on Mac yet
+# net6+ isn't supported on Mac yet
```
Comment (chlafreniere) [inline builds/azure-pipelines/template-steps-build-test.yml:177]: Is .net core 2.1 really still needed for mac?

Comment (Charles-Gagnon) [reply builds/azure-pipelines/template-steps-build-test.yml:177]: I'll check once I've gotten the other issues fixed 😄

**Comment by Charles-Gagnon** [comment]
> Is there an existing issue that this fixes?

No, just was reminded of this from the open PR in STS. We should generally be keeping up to date with the latest SDK on a regular basis - there isn't really any reason we wouldn't want to and it makes sure we get the latest improvements they've made to the build tools.

**Comment by Charles-Gagnon** [comment]
Ok, well turns out that we'd need to install net6 in order to run the tests - which while not difficult is just extra stuff I'd rather not add at this point. So closing this and just doing a separate PR to fix the one code suggestion (since that was something I hit when running test functions that directly referenced the project, and is a good suggestion anyways).

**Pull request closed without merging**